app/src/main/java/es/wolfi/utils/PasswordGenerator.java

```diff
@@ -0,0 +1,142 @@
+package es.wolfi.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+import java.util.Random;
+
+import es.wolfi.app.passman.SettingValues;
+
+public class PasswordGenerator {
+    private static JSONObject getDefaultPasswordGeneratorSettings() throws JSONException {
+        JSONObject passwordGeneratorSettings = new JSONObject();
+
+        passwordGeneratorSettings.put("length", 12);
+        passwordGeneratorSettings.put("useUppercase", true);
+        passwordGeneratorSettings.put("useLowercase", true);
+        passwordGeneratorSettings.put("useDigits", true);
+        passwordGeneratorSettings.put("useSpecialChars", true);
+        passwordGeneratorSettings.put("avoidAmbiguousCharacters", true);
+        passwordGeneratorSettings.put("requireEveryCharType", true);
+
+        return passwordGeneratorSettings;
+    }
+
+    public static JSONObject getPasswordGeneratorSettings(Context context) throws JSONException {
+        SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(context);
+        JSONObject passwordGeneratorSettings = getDefaultPasswordGeneratorSettings();
+
+        try {
+            String cs = settings.getString(SettingValues.PASSWORD_GENERATOR_SETTINGS.toString(), null);
+            if (cs != null) {
+                JSONObject customPasswordGeneratorSettings = new JSONObject(cs);
```

Comment:
Wouldn't it be more efficient to just return the decoded data if a valid json has been found and the defaults otherwise? altough your approach adds flexibility to add new settings in the future. Pick what you think it's better for the job at hand

Reply:
removed with a573e6b4ee82ca0854774161356bf6cb9c9da289